pyo3-macros/src/lib.rs

```diff
@@ -160,6 +161,38 @@ pub fn derive_from_py_object(item: TokenStream) -> TokenStream {
     .into()
 }
 
+#[proc_macro_derive(IntoPyDict, attributes(pyo3, into_py_dict_ignore))]
+pub fn derive_into_pydict(item: TokenStream) -> TokenStream {
+    let cloned = item.clone();
+    let ast = parse_macro_input!(cloned as DeriveInput);
+    check_type(&ast).unwrap();
+    let ident = ast.ident.into_token_stream();
+    let clause_wrapped = ast.generics.where_clause.clone();
+    let mut where_clause: TokenStream2 = TokenStream2::new();
+    let generic_params: TokenStream2 = parse_generics(&ast.generics).parse().unwrap();
+    let generics = ast.generics.into_token_stream();
+
+    if let Some(clause) = clause_wrapped {
+        where_clause = clause.into_token_stream();
+    }
+    let mut dict_fields: Pyo3Collection = Pyo3Collection(Vec::new());
+    for token in item {
+        let token_stream: syn::__private::TokenStream = token.into();
+        dict_fields += parse_macro_input!(token_stream as Pyo3Collection);
+    }
```

Comment:
Don't go back and forth between `syn` types, tokenstreams and strings here and elsewhere. Parse it into `DeriveInput`, *once*, then just write functions that take that or the types contained in it.